testing/tests/DevExpress.ui.widgets.pivotGrid/pivotGrid.tests.js

```diff
@@ -6206,6 +6227,141 @@ QUnit.module('Vertical headers', {
+                        d.resolve({ data: [{ row: 'row', column: 'col', amount: 1 }] });
+                        return d.promise();
+                    }
+                }
+            }).dxPivotGrid('instance');
+            this.clock.tick();
+
+            checkPivotGridCellsText(grid, assert);
+        });
+
+        QUnit.test('PivotGrid. [col x row x data]. remoteOperations = true. Resolve data as arguments list', function(assert) {
+            const grid = $('#pivotGrid').dxPivotGrid({
+                dataSource: {
+                    remoteOperations: true,
+                    fields: [
+                        { area: 'row', dataField: 'row', dataType: 'string' },
+                        { area: 'column', dataField: 'column', dataType: 'string' },
+                        { area: 'data', dataField: 'amount', dataType: 'number', summaryType: 'sum' }
+                    ],
+                    load: function(options) {
+                        const d = $.Deferred();
+                        if(options.group === undefined) {
+                            d.resolve([]); // retrieve fields
+                        } else if(options.group && options.group[0].selector === 'column') {
+                            d.resolve([{ key: 'col', summary: [1] }]);
+                        } else if(options.group && options.group[0].selector === 'row') {
+                            d.resolve([ { key: 'row', summary: [1], items: [ { key: 'col', summary: [1] } ] }], { summary: [1] });
+                        }
+                        return d.promise();
+                    }
+                }
+            }).dxPivotGrid('instance');
+            this.clock.tick();
+
+            checkPivotGridCellsText(grid, assert);
+        });
+
+        QUnit.test('PivotGrid. [col x row x data]. remoteOperations = true. Resolve data as object', function(assert) {
+            const grid = $('#pivotGrid').dxPivotGrid({
+                dataSource: {
+                    remoteOperations: true,
+                    fields: [
+                        { area: 'row', dataField: 'row', dataType: 'string' },
+                        { area: 'column', dataField: 'column', dataType: 'string' },
+                        { area: 'data', dataField: 'amount', dataType: 'number', summaryType: 'sum' }
+                    ],
+                    load: function(options) {
+                        const d = $.Deferred();
+                        if(options.group === undefined) {
+                            d.resolve({ data: [] }); // retrieve fields
+                        } else if(options.group && options.group[0].selector === 'column') {
+                            d.resolve({ data: [{ key: 'col', summary: [1] }] });
+                        } else if(options.group && options.group[0].selector === 'row') {
+                            d.resolve({ data: [{ key: 'row', summary: [1], items: [ { key: 'col', summary: [1] } ] }], summary: [1] });
```

Comment:
Can we use native Promise in one of tests? (with async and setTimeout)

Reply:
Yep, great idea https://github.com/DevExpress/DevExtreme/pull/17577/commits/114ce164c721f9c1f7b3a95b73d2b18e9a705610